candle-book/src/tracing.md

```diff
@@ -0,0 +1,51 @@
+# Tracing
+
+Tracing is a powerful tool for identifying performance issues and bottlenecks in code.
+
+> When using CUDA, Metal, or other devices with asynchronous execution, be aware that tracing might not accurately identify performance issues. GPU operations are typically queued and executed asynchronously, which can lead to misleading timing information in traces.
+> 
+> If are debugging CUDA specifically, you can set the environment variable `CUDA_LAUNCH_BLOCKING=1` which forces synchronous execution. This makes trace timings more accurate, at the cost of reduced performance.
```

Comment:
`CUDA_LAUNCH_BLOCKING=1` should be mentioned but I don't think we should recommend using it + tracing to do performance analysis of the code as the traces are commonly very different from what you would see in a non-blocking world. Instead one should probably use nvidia `nsys profile` command + `nsys-ui`. This produces some profiles for async cuda executions using the cuda kernel names.